src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -343,10 +343,15 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
       }
     }
 
-    const loadBucketLocation = withErrorIgnoring(async (googleProject, bucketName) => {
-      if (!!googleProject) {
+    const loadBucketLocation = withErrorIgnoring(async (googleProject, bucketName, workspace) => {
+      if (isGoogleWorkspace(workspace) && !!googleProject) {
+        // Google
         const bucketLocation = await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketLocation(googleProject, bucketName)
         setBucketLocation(bucketLocation)
+      } else if (isAzureWorkspace(workspace)) {
+        // Azure: region is armRegionName such as eastus
+        const { location } = await Ajax(signal).AzureStorage.details(workspace.workspace.workspaceId)
```

Comment:
Unfortunately this code is problematic, both for Google and Azure workspaces.

For Azure, when a workspace is first cloned, the storage container doesn't yet exist. This means that the `details` endpoint will throw an error. The WorkspaceDashboard handles this by [catching the error and polling](https://github.com/DataBiosphere/terra-ui/blob/a99f98cc08e7cc505361e72a86fe1ebcbdfbc540/src/pages/workspaces/workspace/Dashboard.js#L351). If we want to make the container region something that is passed down through wrapWorkspace, we will have to move that retrying logic up into WorkspaceContainer. The other option is that folks who need the region call the `details` endpoint themselves, and handle the fact that it may error (the file browser calls the `details` endpoint directly and handles the error, for example).

For Google workspaces, the `checkBucketLocation` method will error while IAM propagation is still underway (which can be quite extended, like hours). Handling this more gracefully is something I'm working on now.

Note that this method has `withErrorIgnoring`, which means that if an error is thrown from any of the Ajax methods, the failure will be silent. Users won't get a notification of the failure, but the bucket location also will not be set. 

I'm still trying to think through the best way to handle these transient but expected errors. Can you point out in which locations you are referencing the bucketLocation?

Reply:
Also re-using "bucketLocation" for an Azure workspace is a bit yucky. "Bucket" has always been very much a Google concept, and there's also "bucketName". We may wish to create cloud-specific structures for storing this information.

Reply:
Also, another reason why retrieving the bucketLocation may fail for GCP workspaces is if you are in a requester pays workspace. There is special handling for that case in `WorkspaceDashboard` as well. :/

Reply:
Thinking about this more, for the purposes of this PR I think it is OK to go with the approach here (with some simplifications suggested below), as long as your code that consumes the bucket location is robust to the fact that it may be undefined, regardless of the state of the other workspace details. It would be undefined in the situations I outlined above-- Google workspace where IAM propagation hasn't finished, Google requester pays workspace, Azure newly cloned workspace. You would have to somehow indicate that the location is unknown, and the only way the variable will become populated is if the user refreshes the browser and the workspace is now in a state where fetching the information succeeds.

We need to keep this code using `withErrorIgnoring` for now, and I can try to work out improvements in the ticket I'm working on so that for Azure workspaces and Google workspaces where IAM propagation is delayed, we continue polling and eventually the bucket location will be populated (though again, your code must handle that intermediary state of the variable being undefined); for requester pays workspaces, bucket location will never be available until/if they link to their own billing account.

Simplifications-- since you are passing `workspace` to `setBucketLocation`, you don't  need to pass in `googleProject` and `bucketName`. Instead you can get them directly from the workspace. And if it's a Google workspace, you can expect that `googleProject` is truthy (no need to explicitly check it).

Reply:
Thank you so much for the thorough feedback! I'd like to incorporate as much of this as possible; I'd prefer not to further entangle the code with these changes.

Reply:
For this PR, I think you should incorporate the simplification I called out and ensure that your code that consumes the bucket location is robust to the fact that it may be undefined, regardless of the state of the other workspace details.